### PR TITLE
infra: move all launch.groovy usages in CI to diff.groovy

### PR DESCRIPTION
**Pull request by nmancus1**
infra: move all launch.groovy usages in CI to diff.groovy

Right now this PR is just to test and make sure that all CI works as expected.  See https://github.com/checkstyle/contribution/pull/546 for details.

### CODE REVIEW

File: .ci/no-exception-test.sh [hunk 1]
```diff
@@ -5,16 +5,30 @@ removeFolderWithProtectedFiles() {
...
 
 guava-with-google-checks)
   CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
   echo CS_version: $CS_POM_VERSION
-  mkdir -p .ci-temp/
-  cd .ci-temp/
-  git clone https://github.com/checkstyle/contribution
-  cd contribution/checkstyle-tester
+  checkout_from "-b issue-529 https://github.com/nmancus1/contribution.git"
```
Comment (rnveach) [inline .ci/no-exception-test.sh:30]: These changes seem unrelated to switching from launch to diff. I would move them to a new PR for faster merging as the swap from launch to diff will take some time.

Comment (rnveach) [reply .ci/no-exception-test.sh:30]: I am just now see your repo, so now I understand why the `-b` was added. My bad.

The addition of `checkout_from` seems beneficial, so it makes sense to me to have that in another PR.

**Comment by nmancus1** [comment]
circleci failure: `Caused by: java.net.SocketException: Connection reset`

File: .ci/no-exception-test.sh (hunk 1 above)
Comment (nmancus1) [reply .ci/no-exception-test.sh:11]: Done at https://github.com/checkstyle/checkstyle/pull/9306

**Comment by nmancus1** [comment]
Github, rebase

File: .ci/no-exception-test.sh [hunk 2]
```diff
@@ -59,7 +63,7 @@ openjdk14-with-checks-nonjavadoc-error)
       --mode single \
       --patchConfig checks-nonjavadoc-error.xml \
       --localGitRepo  "$LOCAL_GIT_REPO" \
-      --patchBranch "$BRANCH"
+      --patchBranch "$BRANCH" -xm "Dcheckstyle.failsOnError=false"
```
Comment (romani) [inline .ci/no-exception-test.sh:66]: please keep `-` in front of `D` to make it clear that it is just argument to transfer CLI parameters to inner mvn.

**Review by romani — APPROVED** [review]
this PR need to be updated to use upstream contribution repo as contribution PR is merged.

item:

File: .ci/no-exception-test.sh (hunk 2 above)
Comment (nmancus1) [reply .ci/no-exception-test.sh]: Done.

**Review by romani — APPROVED** [review]
